### PR TITLE
Change ngraph public api

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - PR includes:
    - https://github.com/openvinotoolkit/openvino/pull/6923
    - ngraph::Function->ov::Function
    - ngraph::Variant ->  ov::Variant

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: ngraph/core/include/openvino/core/core_visibility.hpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+// NGRAPH_API is used for the public API symbols. It either DLL imports or DLL exports
+//    (or does nothing for static build)
+
+#ifdef _WIN32
+#    pragma warning(disable : 4251)
+#    pragma warning(disable : 4275)
+#endif
+
+#ifdef NGRAPH_STATIC_LIBRARY  // defined if we are building or calling NGRAPH as a static library
+#    define CORE_API
```
Comment (apankratovantonp) [inline ngraph/core/include/openvino/core/core_visibility.hpp:18]: `CORE_API` -> `OV_CORE_API` or it should be `OV_API`

File: ngraph/core/include/ngraph/ngraph_visibility.hpp [hunk 2]
```diff
@@ -3,32 +3,6 @@
...
-#ifndef ENABLE_UNICODE_PATH_SUPPORT
-#    ifdef _WIN32
-#        if defined __INTEL_COMPILER || defined _MSC_VER
-#            define ENABLE_UNICODE_PATH_SUPPORT
-#        endif
-#    elif defined(__GNUC__) && (__GNUC__ > 5 || (__GNUC__ == 5 && __GNUC_MINOR__ > 2)) || defined(__clang__)
-#        define ENABLE_UNICODE_PATH_SUPPORT
-#    endif
-#endif
+#define NGRAPH_API CORE_API
```
Comment (ilya-lavrenov) [inline ngraph/core/include/ngraph/ngraph_visibility.hpp:8]: `OPENVINO_API`?

File: ngraph/core/include/ngraph/variant.hpp [hunk 3]
```diff
@@ -7,90 +7,30 @@
 #include <map>
 #include <string>
```
Comment (ilya-lavrenov) [inline ngraph/core/include/ngraph/variant.hpp:8]: do we need these headers?

File: ngraph/core/include/openvino/core/visibility.hpp [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+#    define CORE_HELPER_DLL_LOCAL
+#elif defined(__GNUC__) && __GNUC__ >= 4
+#    define CORE_HELPER_DLL_IMPORT __attribute__((visibility("default")))
+#    define CORE_HELPER_DLL_EXPORT __attribute__((visibility("default")))
+#    define CORE_HELPER_DLL_LOCAL  __attribute__((visibility("hidden")))
+#else
+#    define CORE_HELPER_DLL_IMPORT
+#    define CORE_HELPER_DLL_EXPORT
+#    define CORE_HELPER_DLL_LOCAL
+#endif
```
Comment (ilya-lavrenov) [inline ngraph/core/include/openvino/core/visibility.hpp:19]: should not we copy all stuff from `ie_api.hpp`?

File: ngraph/frontend/frontend_manager/include/frontend_manager/frontend_manager.hpp [hunk 5]
```diff
@@ -93,6 +93,9 @@ struct FrontEndPluginInfo {
 };
 
 }  // namespace frontend
+}  // namespace ngraph
```
Comment (ilya-lavrenov) [inline ngraph/frontend/frontend_manager/include/frontend_manager/frontend_manager.hpp:96]: BTW, I think we have move all frontend related stuff to OV namespace w/o compatibility headers.

They were not released

**Review by ilya-lavrenov — APPROVED** [review]

File: ngraph/frontend/frontend_manager/include/frontend_manager/frontend_manager.hpp (hunk 5 above)
Comment (ilyachur) [reply ngraph/frontend/frontend_manager/include/frontend_manager/frontend_manager.hpp:96]: I would like to do it in the separate PR

File: ngraph/core/include/ngraph/variant.hpp (hunk 3 above)
Comment (ilyachur) [reply ngraph/core/include/ngraph/variant.hpp:8]: yes we need. We use both structures in this header

File: ngraph/core/include/ngraph/ngraph_visibility.hpp (hunk 2 above)
Comment (ilyachur) [reply ngraph/core/include/ngraph/ngraph_visibility.hpp:8]: Changed

File: ngraph/core/include/openvino/core/visibility.hpp (hunk 4 above)
Comment (ilyachur) [reply ngraph/core/include/openvino/core/visibility.hpp:19]: We have some duplication of these structure, I propose to do it separately, the idea of this PR is the creation of entry point for changing nGraph API.

File: ngraph/core/include/openvino/core/function.hpp [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+#include "ngraph/op/result.hpp"
+#include "ngraph/op/sink.hpp"
+#include "ngraph/op/util/variable.hpp"
+#include "openvino/core/core_visibility.hpp"
+
+namespace ov {
+/// A user-defined function.
+class OPENVINO_API Function {
+public:
+    static constexpr ngraph::DiscreteTypeInfo type_info{"Function", 0};
```
Comment (apankratovantonp) [inline ngraph/core/include/openvino/core/function.hpp:27]: There were some ngraph type_info declaration MACRO. Should we use them here?

**Review by apankratovantonp — APPROVED** [review]

**Pull request merged**